packages/plugin-dom-follow-range/src/FollowRangeZoneDomObjectRenderer.ts

```diff
@@ -0,0 +1,95 @@
+import {
+    DomObjectRenderingEngine,
+    DomObject,
+} from '../../plugin-html/src/DomObjectRenderingEngine';
+import { AbstractRenderer } from '../../plugin-renderer/src/AbstractRenderer';
+import { FollowRangeZoneNode } from './FollowRangeZoneNode';
+
+import '../assets/FollowRange.css';
+import { DomLayoutEngine } from '../../plugin-dom-layout/src/ui/DomLayoutEngine';
+import { Layout } from '../../plugin-layout/src/Layout';
+import { RenderingEngine, Renderer } from '../../plugin-renderer/src/RenderingEngine';
+
+export class FollowRangeZoneDomObjectRenderer extends AbstractRenderer<DomObject> {
+    static id = DomObjectRenderingEngine.id;
+    engine: DomObjectRenderingEngine;
+    predicate = FollowRangeZoneNode;
+
+    private _nbAttached = 0;
+    private _hide = true;
+    private readonly _followRangeDebounced: () => void;
+
+    constructor(engine: RenderingEngine<DomObject>, superRenderer: Renderer<DomObject>) {
+        super(engine, superRenderer);
+        let debounce: number;
+        this._followRangeDebounced = (): void => {
+            window.clearTimeout(debounce);
+            debounce = window.setTimeout(this._followRange.bind(this), 3);
+        };
+    }
+
+    async render(node: FollowRangeZoneNode): Promise<DomObject> {
+        if (node.hasChildren()) {
+            const followRange: DomObject = {
+                tag: 'JW-FOLLOW-RANGE',
+                children: node.children(),
+                attach: (el: HTMLElement): void => {
+                    if (this._hide) {
+                        el.style.display = 'none';
+                    }
+                    this._nbAttached++;
+                    if (this._nbAttached === 1) {
+                        document.addEventListener(
+                            'selectionchange',
+                            this._followRangeDebounced,
+                            true,
+                        );
+                    }
+                },
+                detach: (el: HTMLElement): void => {
+                    el.style.display = 'none';
+                    this._nbAttached--;
+                    if (this._nbAttached === 0) {
+                        this._hide = true;
+                        document.removeEventListener(
+                            'selectionchange',
+                            this._followRangeDebounced,
+                            true,
+                        );
+                    }
+                },
+            };
+            return followRange;
+        } else {
+            return { children: [] };
+        }
+    }
+
+    private _followRange(): void {
+        const selection = window.getSelection();
+        const display =
+            selection.anchorNode !== selection.focusNode ||
+            selection.anchorOffset !== selection.focusOffset;
+        if (display || !this._hide) {
```

Comment:
Je ne comprends pas le lien entre display et this._hide.

Reply:
Si ne n'est pas visible et que ça ne doit pas être affiché on ne fait rien. Si c'est affiché on modifie la position.